sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/renderers/HierarchyTreeNodeSelectRenderer.java

```diff
@@ -181,37 +181,88 @@ public void renderSelectHierarchyNodesTree(UIContainer parent, String clientID,
-				   Set<String> currentNodeParents = node.parentNodeIds;
-				   currentNodeParents.add(node.id);
-				   selectedGroups.remove(groupID);
-				   renderRow(tofill, "hierarchy-level-row:", level+1, evalGroupObj, evalViewParams, accessNodeIds, currentNodeParents, false);
-			   }
-    		   for (EvalHierarchyNode childHierNode: hierarchyLogic.getChildNodes(node.id, true)) {
-    			   renderSelectHierarchyNode(tofill, childHierNode, level+1, evalViewParams, accessNodeIds, parentNodeIds, selectedNodes, selectedGroups);
-    		   }
-    	   }
-       }
+    /**
+     * This is the main entry point for rendering the hierarchy with selectable
+     * checkboxes.  The parent container should be inside whatever form is being
+     * used. The groups and nodes SelectID's are the getFullID's that were generated
+     * from the UISelect. We're doing a GET form here so there are no EL Bindings
+     * for these checkboxes.  The Lists of things are used for storing the option
+     * label/value at each point.  These are live references that are passed in 
+     * and used after this method invocation to update the UISelect.optionslist
+     * and UISelect.optionnames for the Hierarchy Nodes and Group Nodes.
+     * 
+     * From a visual perspective, the Hierarchy and Group Nodes are interspersed
+     * together throughout the page, but we keep them seperate with the two 
+     * UISelects.
+     * 
+     * @param parent
+     * @param clientID
+     * @param groupsSelectID
+     * @param hierNodesSelectID
+     * @param evalGroupLabels
+     * @param evalGroupValues
+     * @param hierNodeLabels
+     * @param hierNodeValues
+     * @param evalViewParams
+     * @param accessNodeIds
+     * @param parentNodeIds
+     * @param sectionAware
+     */
+    public void renderSelectHierarchyNodesTree( UIContainer parent, String clientID, String groupsSelectID, String hierNodesSelectID,
+             List<String> evalGroupLabels, List<String> evalGroupValues, List<String> hierNodeLabels, List<String> hierNodeValues, 
+             EvalViewParameters evalViewParams, Set<String> accessNodeIds, Set<String> parentNodeIds, boolean sectionAware )
+    {
+        this.sectionAware = sectionAware;
+        renderSelectHierarchyNodesTree( parent, clientID, groupsSelectID, hierNodesSelectID, evalGroupLabels, evalGroupValues, hierNodeLabels,
+                hierNodeValues, evalViewParams, accessNodeIds, parentNodeIds );
     }
-    
+
+    /**
+     * Performs the recursive rendering logic for a single hierarchy node.
+     * 
+     * @param tofill
+     * @param node
+     * @param level
+     */
+    private void renderSelectHierarchyNode(UIContainer tofill, EvalHierarchyNode node, int level, EvalViewParameters evalViewParams, Set<String> accessNodeIds, Set<String> parentNodeIds,
+           List<String> selectedNodes, List<String> selectedGroups) {
+
+        //a null "accessNodeIds varaible means the user is admin
```

Comment:
Not necessary here, as it's only being passed to the `renderRow()` method, which does the null checking on it.